### DISCUSSION TITLE
Welcome to BezierEditing Discussions!

### DISCUSSION THREAD

**Discussion by tmizu23** [tmizu23/BezierEditing#9] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by SumitKalyan** [comment]
Hi, 

This tool is extremely useful in my work. I just wanted to know if there is a way to move both the handles of a vertex with respect to each other once the vertex is created?

Thanks

**Comment by tmizu23** [comment]
To move the handle, click and hold, then drag.
Is that what you mean?

thanks

**Comment by SumitKalyan** [comment]
Thank you for your reply. 

Yes, but one should move with respect to the other handle for that particular vertex, meaning an option to keep them connected. Please see the gif attached for clarification. 

![Handle_drag](https://user-images.githubusercontent.com/79241827/157136212-732200b2-62ea-4797-b1cf-8bdc2af4b963.gif)


I understand the fact that making an edit or addition to a plugin is highly complex but another thing that I struggle with this tool is sharp edges. Please see the image attached. Is there any way to cut a large line segment into smaller pieces to make it look a bit smoother? This change would be highly helpful even more than the above-mentioned function. 

![sharp](https://user-images.githubusercontent.com/79241827/157149474-ad232fbc-327d-4aea-bc5b-711a8e5bb7bd.png)

I know there are some smoothing algorithms out there but after applying those my feature is not bezier anymore and I get the following message which destroys the feature completely. 

![image](https://user-images.githubusercontent.com/79241827/157146507-cdac2b12-35aa-43e7-825b-dd82bbe20745.png)

Thanks

**Reply by tmizu23** [reply]
I've implemented it, can you test it?
Click here for installation instructions.

https://github.com/tmizu23/BezierEditing/discussions/22#discussioncomment-2314196

**Comment by SumitKalyan** [comment]
Thank you for the changes. The number of interpolations setting works perfectly. It really made a huge difference in the output of my work. 

I have a suggestion though. Since we can not change the number of interpolations in the middle of drawing a feature, this causes an issue of consistency in the smoothness (see the image below). I used interpolate point count = 20 throughout. 
![image](https://user-images.githubusercontent.com/79241827/157358922-05cea18a-47db-4d2b-adf4-61f59c10cf68.png)
Is there a way to make the interpolation number depended on the length of the segment like 1 point every 2 metes or 1 point every 5 meters instead of a fixed interpolate point count between them? 

Just to give an idea the pseudo-code for this could be something like:
Get coordinates of both anchors Let's say **x₁y₁ and x₂y₂**
Calculate the distance "**D= √[(x₂ - x₁)² + (y₂ - y₁)²]**" lets assume it comes out 50 
User input for number of points "**points interval" = ___ meters**" Lets assume users enteres 2
"**Interpolation point count = D/points interval**" i.e 100/2 = 50 

Thanks again for the edits.
Regards,

**Reply by tmizu23** [reply]
Thanks for the suggestion.
Your method works when creating. However, this tool does not store which points are anchor points, so if the number of interpolations varies, it will no longer be able to calculate which points are anchor points. Therefore, you will not be able to edit the curve as a bezier curve again.
This would be possible if the anchor points could be saved, but I do not plan to do so.

Do you have any other good ideas?

**Comment by SumitKalyan** [comment]
Oh, I see. All good. The only suggestion that I can think of is to have an option to keep both the handles of an anchor locked with each other. 

In the following gif, the handles of an anchor are locked while creating a new anchor but not sure if there is an option to keep them locked after the creation as well
![ezgif com-gif-maker](https://user-images.githubusercontent.com/79241827/159380756-7afbfdfd-0106-45c7-8cf1-dbac1db6dc76.gif)

Thank you for your support.

**Reply by tmizu23** [reply]
I've implemented it.
download v1.3.3beta3 from here
[https://github.com/tmizu23/BezierEditing/releases](https://github.com/tmizu23/BezierEditing/releases)

**Reply by SumitKalyan** [reply]
Thanks a lot for the edits. The tool is and will be highly useful. 

There is a minor thing about the new option of connected handles though. They struggle to snap or move around the straight lines. In this following gif, firstly, I moved the single handle (disconnected with each other) around the straight segment which worked perfectly but then tried the same thing with connected handles and it stops moving in that area. 
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/79241827/160059927-821671e9-3d0c-400d-9b3e-690a875f2828.gif)

Thank you again for the effort that you put into creating and improving this Plugin.

**Reply by tmizu23** [reply]
Thank you for your report. I fix it. 
Please test v1.3.3beta4

**Comment by DL-Works** [comment]
I just needed to thank you for this plugin, it has proved to be critical in my work!
Thank you again.
@tmizu23

**Reply by tmizu23** [reply]
You're very welcome! I'm really glad to hear it's been helpful in your work.
